### PR TITLE
[Feature] Introduce new unit tests for error info view model 

### PR DESCRIPTION
**Pull request by JoshuaLai**
## Purpose
Introducing new unit tests for the error info view. Taking the localization into consideration. 

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUITests/Presentation/Setup/ErrorInfoViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        localizationProvier = LocalizationProviderMocking()
+    }
+
+    func test_dismissContent_alwaysReturns_snackBarDismissCintent() {
+        let sut = makeSUT()
+
+        XCTAssertEqual(sut.dismissContent, "AzureCommunicationUI.SnackBar.Button.Dismiss")
+    }
+
+    func test_errorCodeCallJoin_returns_snackBarErrorJoinCallMessage() {
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUITests/Presentation/Setup/ErrorInfoViewModelTests.swift:26]: Could you update functions naming to follow the common pattern?

**Review by vhuseinova-msft — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/AzureCommunicationUITests/Presentation/Setup/ErrorInfoViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private var localizationProvier: LocalizationProviderMocking!
+    private let timeout: TimeInterval = 10.0
+
+    override func setUp() {
+        super.setUp()
+        storeFactory = StoreFactoryMocking()
+        localizationProvier = LocalizationProviderMocking()
+    }
+
+    func test_dismissContent_alwaysReturns_snackBarDismissCintent() {
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUITests/Presentation/Setup/ErrorInfoViewModelTests.swift:20]: minor typo (Cintent -> Content)

**Review by vhuseinova-msft — DISMISSED** [review]

File: AzureCommunicationUI/AzureCommunicationUITests/Presentation/Setup/ErrorInfoViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    func test_errorInfoViewModel_update_when__errorStateCallEnd_then_snackBarErrorCallEndMessage() {
+        let sut = makeSUT()
+        let event = ErrorEvent(code: CallCompositeErrorCode.callEnd)
+        let state = ErrorState(error: event, errorCategory: .callState)
+
+        sut.update(errorState: state)
+        XCTAssertEqual(sut.isDisplayed, true)
+        XCTAssertEqual(sut.message, "AzureCommunicationUI.SnackBar.Text.ErrorCallEnd")
+    }
```
Comment (AmyL219) [inline AzureCommunicationUI/AzureCommunicationUITests/Presentation/Setup/ErrorInfoViewModelTests.swift:41]: Have we tested when the errorType is update from an Error to empty, the `isDisplay` would be false?
Generally speaking, we would hide the error banner if the error type become `empty` when we restart the call

**Review by jimchou-dev — APPROVED** [review]
LGTM

**Review by AmyL219 — APPROVED** [review]
Looks good!

**Pull request merged**